### PR TITLE
Subtitle rendering

### PR DESCRIPTION
**Pull request by elysoly**

### CODE REVIEW

**Comment by NaveenRagunathan** [comment]
Hi @elysoly can you elaborate on this subtitle rendering issue?

File: tests/test_postprocessing.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+from vikit.common.context_managers import WorkingFolderContext
+from vikit.postprocessing.video_subtitle_renderer import VideoSubtitleRenderer
+
+PARIS_VIDEO = get_paris_video()
+SUBTITLE_PATH = get_paris_subtitle_file()
+
+
+class TestPostProcessing:
+    def setUp(self) -> None:
+        warnings.simplefilter("ignore", category=ResourceWarning)
```
Comment (jeffmac-aix) [inline tests/test_postprocessing.py:31]: Now we use to add warnings and log file definition before the class declaration, not working in setup

**Review by jeffmac-aix — COMMENTED** [review]
tests/test_postprocessing.py::TestPostProcessing::test_render_subtitle  does not pass for me, though it passed with CI but CI does not ship all our local integration tests. Could you check from a fresh venv?

**Comment by elysoly** [comment]
> Hi @elysoly can you elaborate on this subtitle rendering issue?

Hello @NaveenRagunathan :) 
I made an issue about it: [issue 123](https://github.com/Vikit-ai/sdk/issues/123#issue-2458984273). 
Overall it is working with this PR, the only issue is that the duration of the video with subtitles is a few milliseconds longer than the original one. For example, the original video is 15.05 sec long, while the subtitled version is 15.08 sec. I think the problem is coming from ffmpeg parameters, because when I just rewrite the original video (without subtitles and audio), I still have the same problem. We can  modify ffmpeg parameters:
```python
 final_video.write_videofile(
            output_video_path,
            fps=video.fps,
            codec=self.codec,
            audio_codec="aac",
            ffmpeg_params="",
        )
```

**Comment by elysoly** [comment]
> tests/test_postprocessing.py::TestPostProcessing::test_render_subtitle does not pass for me, though it passed with CI but CI does not ship all our local integration tests. Could you check from a fresh venv?

It passes for me. What is the error you get?

File: tests/test_postprocessing.py (hunk 1 above)
Comment (elysoly) [reply tests/test_postprocessing.py]: I pushed the fix. just that I copied from other test files ( /tests/test_video.py should be fixed there as well maybe)

**Review by elysoly — COMMENTED** [review]
Just need an update about ImageMagick installation on Readme @leclem

**Pull request merged**